node/node.go

```diff
@@ -1077,6 +1039,69 @@ func (node *Node) setupProtocols() error {
 	return nil
 }
 
+// CreateMiningWorker creates a mining.Worker for the node using the configured
+// getStateTree, getWeight, and getAncestors functions for the node
+func (node *Node) CreateMiningWorker(ctx context.Context) (mining.Worker, error) {
+	processor := consensus.NewDefaultProcessor()
+
+	minerAddr, err := node.miningAddress()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get mining address")
+	}
+
+	minerPubKey, err := node.PorcelainAPI.MinerGetKey(ctx, minerAddr)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get key from miner actor")
+	}
+
+	minerOwnerAddr, err := node.miningOwnerAddress(ctx, minerAddr)
+	if err != nil {
+		log.Errorf("could not getKey from miner actor")
+		return nil, err
+	}
+	return mining.NewDefaultWorker(
+		node.MsgPool, node.getStateTree, node.getWeight, node.getAncestors, processor, node.PowerTable,
```

Comment:
--Disclaimer-- Not at all suggesting this change in this PR.

The miner worker is clearly a consumer of a lot of core pieces.  If we raise its logic up into the block mining protocol then we can probably hit all of these dependencies through the porcelain api.  This might open up a way to get rid of the need to use these somewhat clunky functions for initializing the worker.  Instead the blockMiner (previously some combo of scheduler and worker), as defined in `protocol/block_mining.go`  could be modularly tested by using the carefully designed mocking features of the porcelain.

Reply:
In retrospect this is probably exactly what you are already thinking as moving the mining api to `protocol/block` seems to imply a blockMiner by symmetry with `protocol/storage`'s storageMiner and `protocol/retrieval`'s retrievalMiner.